[Arcane Mage] The War Within Update

Sharrq opened the pull request:
The War Within Update for Arcane Mage, pretty much a complete rewrite.

- Added Guide view
- Started removing Checklist and Suggestions
- Rewrote several modules including Arcane Surge, Touch of the Magi, Clearcasting, Arcane Orb, Arcane Barrage, Nether Precision, Siphon Storm, and Arcane Tempo.
- Added Arcane Tempo to Haste
- Added Mana Level graph to Guide
- Updated spec support to 11.0.0
- Added a warning regarding prepatch
- Removed Checklist from Log Parser so it cant be loaded anymore
![Heroic-Igira-the-Cruel-Kill-2-31-by-Andarion-in-Aberrus-the-Shadowed-Crucible (1)](https://github.com/user-attachments/assets/89b37fe1-2082-4e96-8f2d-32bab8928241)

Sharrq commented on src/interface/report/Results/Overview.tsx:
@emallson I made this change to core so alerts in config get showed on the Guide view too. I added the warning for Arcane and it was only showing on the Checklist view. Everything else in this PR is spec specific and I'm going to merge it myself ... just wanted to make sure you were cool with this one change to interface/report/Results/Overview.tsx

kfinch reviewed (commented):
Overall looking good, although I want to pull your branch and play with it myself when I get the chance. Some comments:
* Would be nice to see Evocation / Arcane Surge / Touch of the Magi times on the Mana graph (don't need this in for initial merge, just a thought). I also don't think the boss health overlay actually needs to be there.
* Did the player in your screenshot actually never use Shifting Power? Double check you don't have the wrong cast ID listed for the Cooldown bar.
* Why do some of the cooldowns show icon on cast and others don't? The CDs all seem long enough that they can all show icon.
* I like the breakdown of sections Core > Burn > Rotation > Buffs&Procs - it works well
* The total amount of text should be cut down IMO - can be very daunting to look at as is. When I first made my Resto and Feral Guides they looked very much like this, but over many editing passes I ruthlessly cut down the number of words (frankly Resto needs another pass).
* You can show a mini uptime-graph for ArcaneTempo using the same component I built for showing debuff uptimes. For the simplest version, see `guideSubsection` in the Guardian module called 'Moonfire'
```ts
const history = this.deps.enemies.getDebuffHistory(SPELLS.MOONFIRE_DEBUFF.id);
const uptimeBar = uptimeBarSubStatistic(this.owner.fight, {
  spells: [SPELLS.MOONFIRE_DEBUFF],
  uptimes: history,
  color: MF_COLOR,
});
const data = (
  <div>
    <RoundedPanel>
      <strong>Moonfire uptimes</strong>
      {uptimeBar}
    </RoundedPanel>
  </div>
);
```
(you would use `this.selectedCombatant.getBuffHistory` instead of course)

Sharrq commented:
This pr is not meant to be a finished product. I still have hero talents and such to add which will change some stuff as well. My goal is to get all 3 specs as far as I can before raid launch and then add on stuff after raid launch or before if I have time. The other goal of this is to get SOMETHING in front of the staff from the mage discord so I can start getting feedback.

The mana graph additions would be great. I didn't really know how to do that so unless I can find a clear example of it then it might need to wait until later. 

I'm pretty sure they didn't cast shifting power but I'll double check. I intentionally look up bad logs when I build stuff so that I have an assortment of good and bad, makes it easier to see if stuff is working. 

I copied the fire guide as a starting point and replaced the spell ids on those cool down graphs, probably didn't remember to add the icon back to some of them. 

I know I have a tendency to over word things. For some stuff it's difficult because there are a lot of things for the player to take into account, some of which can be a little complicated, and I want to make sure they understand all the pieces without having to go to some other guide or website to get that important info. I can pare things down, but probably something I'll do towards the end

I will absolutely add that graph for arcane tempo, I know there are other guide elements I could be incorporating too, but I'm still very new to the guide so I haven't figured out how most of them work yet.

Sharrq commented:
@kfinch

- Yes they did not cast Shifting Power
- I added the graph for Arcane Tempo uptime
- I moved the cast efficiency bars into their respective sections above (except Shifting Power which I still need to add)
- Ill wait a bit to do the wording since I have more stuff to add in a second pass and some of that text might change anyway once I add hero talents and such
- idk how to add spells and such to the mana graph, so I'll save that for the next pass

kfinch commented:
Excellent, sounds good to me. I know this isn't supposed to be "done", but it looks like it's already pretty good. I'm of the opinion that we should merge sooner rather than later, and you can add enhancements / hero talent stuff in follow-on PRs. I don't want this PR to sit around for weeks with stuff constantly being added - better for the new stuff to follow in smaller and easier to review PRs.

Sharrq commented:
Yea, I added shifting Power just so I can move that last cast efficiency. Something is weird with arcane Barrage but I can fix that in another pr

Sharrq commented:
actually, I might have found the issue with barrage. so I'll fix that and then merge it
